vcl_templates/frontend.vcl.erb

```diff
@@ -0,0 +1,96 @@
+sub vcl_recv {
+#FASTLY recv
+
+  if (req.request != "HEAD" && req.request != "GET" && req.request != "FASTLYPURGE") {
+    return(pass);
+  }
+
+  if (!req.http.Cookie:ABTest-Example) {
+    if (randombool(5,10)) {
+       set req.http.X-ABTest-Example = "A";
+    } else {
+       set req.http.X-ABTest-Example = "B";
+    }
+  } else {
+    # Set the value of the header to whatever decision was previously made
+    set req.http.X-ABTest-Example =   req.http.Cookie:ABTest-Example;
+  }
+
+  return(lookup);
+}
+
+sub vcl_fetch {
+#FASTLY fetch
+
+  if ((beresp.status == 500 || beresp.status == 503) && req.restarts < 1 && (req.request == "GET" || req.request == "HEAD")) {
+    restart;
+  }
+
+  if (req.restarts > 0) {
+    set beresp.http.Fastly-Restarts = req.restarts;
+  }
+
+  if (beresp.http.Set-Cookie) {
+    set req.http.Fastly-Cachetype = "SETCOOKIE";
+    return(pass);
+  }
+
+  if (beresp.http.Cache-Control ~ "private") {
+    set req.http.Fastly-Cachetype = "PRIVATE";
+    return(pass);
+  }
+
+  if (beresp.status == 500 || beresp.status == 503) {
+    set req.http.Fastly-Cachetype = "ERROR";
+    set beresp.ttl = 1s;
+    set beresp.grace = 5s;
+    return(deliver);
+  }
+
+  if (beresp.http.Expires || beresp.http.Surrogate-Control ~ "max-age" || beresp.http.Cache-Control ~ "(s-maxage|max-age)") {
+    # keep the ttl here
+  } else {
+    # apply the default ttl
+    set beresp.ttl = 3600s;
+  }
+
+  return(deliver);
+}
+
+sub vcl_hit {
+#FASTLY hit
+
+  if (!obj.cacheable) {
+    return(pass);
+  }
+  return(deliver);
+}
+
+sub vcl_miss {
+#FASTLY miss
+  return(fetch);
+}
+
+sub vcl_deliver {
+  # Set the A/B cookie
+  if (req.http.Cookie !~ "ABTest-Example") {
```

Comment:
Sounds sensible to me. I would also rather only have changes in one place to make it simpler. Since we only need to set the cookie once, an extra header would work.